backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/Linker.kt

```diff
@@ -97,4 +102,36 @@ internal class Linker(val context: Context) {
         return executable
     }
 
+}
+
+private class CachesToLink(val static: List<String>, val dynamic: List<String>)
+
+private fun determineCachesToLink(context: Context): CachesToLink {
+    val staticCaches = mutableListOf<String>()
+    val dynamicCaches = mutableListOf<String>()
+
+    // TODO: suboptimal, see e.g. [LlvmImports].
+    context.librariesWithDependencies.forEach { library ->
+        val currentBinaryContainsLibrary = context.llvmModuleSpecification.containsLibrary(library)
+        val cache = context.config.cachedLibraries.getLibraryCache(library)
+        val libraryIsCached = cache != null
+
+        // Consistency check. Generally guaranteed by implementation.
+        if (currentBinaryContainsLibrary && libraryIsCached) {
+            error("Library ${library.libraryName} is found in both cache and current binary")
+        } else if (!currentBinaryContainsLibrary && !libraryIsCached) {
+            error("Library ${library.libraryName} is not found neither in cache nor in current binary")
```

Comment:
Not very good diagnostic if valid relative path is passed.

Reply:
What do you mean?

Reply:
If path is not absolute, cryptic error was shown, IIRC.